tests/core/featurizers/test_tracker_featurizer.py

```diff
@@ -49,48 +55,1010 @@ def test_convert_labels_to_ids(domain: Domain):
 
 
+def test_convert_intent_labels_to_ids(domain: Domain):
+    trackers_as_intents = [
+        ["next_intent", "nlu_fallback", "out_of_scope", "restart"],
+        ["greet", "hello", "affirm"],
+    ]
+
+    tracker_featurizer = IntentMaxHistoryTrackerFeaturizer()
+
+    actual_labels = tracker_featurizer._convert_labels_to_ids(
+        trackers_as_intents, domain
+    )
+
+    expected_labels = np.array([[7, 8, 9, 10], [5, 6, 0, -1]])
+    assert expected_labels.size == actual_labels.size
+    assert expected_labels.shape == actual_labels.shape
+    assert np.all(expected_labels == actual_labels)
+
+
 def test_featurize_trackers_raises_on_missing_state_featurizer(domain: Domain):
     tracker_featurizer = TrackerFeaturizer()
 
     with pytest.raises(ValueError):
         tracker_featurizer.featurize_trackers([], domain, RegexInterpreter())
 
 
+@pytest.fixture
+def moodbot_features(moodbot_domain: Domain) -> Dict[Text, Dict[Text, Features]]:
+    """Makes intent and action features for the moodbot domain to faciliate
+    making expected state features.
+
+    Returns:
+      A dict containing dicts for mapping action and intent names to features.
+    """
+
+    action_shape = (1, len(moodbot_domain.action_names_or_texts))
+    actions = {}
+    for index, action in enumerate(moodbot_domain.action_names_or_texts):
+        actions[action] = Features(
+            sparse.coo_matrix(([1.0], [[0], [index]]), shape=action_shape),
+            FEATURE_TYPE_SENTENCE,
+            ACTION_NAME,
+            "SingleStateFeaturizer",
+        )
+    intent_shape = (1, len(moodbot_domain.intents))
+    intents = {}
+    for index, intent in enumerate(moodbot_domain.intents):
+        intents[intent] = Features(
+            sparse.coo_matrix(([1.0], [[0], [index]]), shape=intent_shape),
+            FEATURE_TYPE_SENTENCE,
+            INTENT,
+            "SingleStateFeaturizer",
+        )
+    return {"intents": intents, "actions": actions}
+
+
+def compare_featurized_states(
+    states1: List[Dict[Text, List[Features]]], states2: List[Dict[Text, List[Features]]]
+) -> bool:
+    """Compares two lists of featurized states and returns True if they
+    are identical and False otherwise.
+    """
+
+    if len(states1) != len(states2):
+        return False
+
+    for state1, state2 in zip(states1, states2):
+        if state1.keys() != state2.keys():
+            return False
+        for key in state1.keys():
+            for feature1, feature2 in zip(state1[key], state2[key]):
+                if np.any((feature1.features != feature2.features).toarray()):
+                    return False
+                if feature1.origin != feature2.origin:
+                    return False
+                if feature1.attribute != feature2.attribute:
+                    return False
+                if feature1.type != feature2.type:
+                    return False
+
+    return True
+
+
+@pytest.fixture
```

Comment:
I think all fixtures should go to `tests/core/conftest.py` to 1. make them available across tests and 2. Have them in one place . @twerkmeister Could you please advise?